src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncIteratorTests.cs

```diff
@@ -6326,6 +6332,58 @@ static async System.Collections.Generic.IAsyncEnumerable<int> Iter(int value, [E
             // IL for GetAsyncEnumerator is verified by AsyncIteratorWithAwaitCompletedAndYield already
         }
 
+        [Fact, WorkItem(34407, "https://github.com/dotnet/roslyn/issues/34407")]
+        public void CancellationTokenParameter_NoTokenPassedInGetAsyncEnumerator_LocalFunction()
```

Comment:
I'd be tempted to add a few more scenarios validating the execution, since the above IL is difficult to review (but I can see that it's doing the token combination as expected).
Consider picking a couple from:
- CancellationTokenParameter_SameTokenPassedInGetAsyncEnumerator
- CancellationTokenParameter_DefaultTokenPassedInGetAsyncEnumerator
- CancellationTokenParameter_SomeTokenPassedInGetAsyncEnumerator
- CancellationTokenParameter_SomeTokenPassedInGetAsyncEnumerator_OptionalParameter
- CancellationTokenParameter_SomeTokenPassedInGetAsyncEnumerator_ButNoAttribute
- CancellationTokenParameter_SomeOtherTokenPassedInGetAsyncEnumerator
- CancellationTokenParameter_TwoDefaultTokens

Reply:
Added some of the test cases you suggested, thanks!

---
In reply to: [366558769](https://github.com/dotnet/roslyn/pull/40959#discussion_r366558769) [](ancestors = 366558769)